### DISCUSSION TITLE
Maximum execution time for Edge functions

### DISCUSSION THREAD

**Discussion by lida-spaceport** [vercel/community#3236] [Feature Question] [answered]
### Question

From my testing, edge function waitUntil will silently fail after 30s.

Take the example code from [Edge documentation](https://vercel.com/docs/concepts/functions/edge-functions/edge-functions-api#waituntil):

```
import { NextResponse } from 'next/server';
import type { NextFetchEvent, NextRequest } from 'next/server';
 
export const config = {
  runtime: 'edge',
};
 
const wait = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
 
async function getAlbum() {
  const res = await fetch('https://jsonplaceholder.typicode.com/albums/1');
  await wait(40000);
  return res.json();
}
 
export default function MyEdgeFunction(
  request: NextRequest,
  context: NextFetchEvent,
) {
  context.waitUntil(getAlbum().then((json) => console.log({ json })));
 
  return NextResponse.json({
    name: `Hello, from ${request.url} I'm an Edge Function!`,
  });
}

```

But change the timeout to 40 seconds and nothing gets logged. So do edge functions only allow more than 30 seconds of execution time if the response is streaming? If so, the documentation should really make it clear that waitUntil has that limitation.

**Comment by amyegan** [comment, marked as answer]
Hey @lida-spaceport. The [maximum initial response time](https://vercel.com/docs/concepts/functions/edge-functions/limitations#maximum-initial-response-time) for Edge Functions is 30 seconds. But that is specific to _initial_ response and [streaming](https://vercel.com/docs/concepts/functions/edge-functions/streaming) is a popular solution for longer execution. If the function begins sending a response within 30 seconds, you can continue streaming a response beyond that time limit.

**Reply by vincentdnl** [reply]
But HOW LONG? It's never said anywhere in the documentation.
